woocommerce/woocommerce
# 27706 Allow Payment Method HTML

Closes #[27706](https://github.com/woocommerce/woocommerce/issues/27706).

### All Submissions:


### Changes proposed in this Pull Request:
Changes escaping method(s) to `wp_kses_post()` to allow markup where the Payment Method title is displayed.  This enables images and links to be used.

This is modified in the following specific places in the code:
* [class-wc-meta-box-order-data.php, ln 206](https://github.com/woocommerce/woocommerce/compare/trunk...johnregan3:add/22706-allow-payment-logo?expand=1#diff-4d6ef31a239517450c07e658510ea400d15787a4d5ace348159994a9e6cd3ed7R206)
* [html-order-items.php, ln 338](https://github.com/woocommerce/woocommerce/compare/trunk...johnregan3:add/22706-allow-payment-logo?expand=1#diff-788a10d345a994fd3a7253a48840dc1dcf4e1466cf03ee0cbc85b53a7d2ae273R338)
* [order-details.php, ln 118](https://github.com/woocommerce/woocommerce/compare/trunk...johnregan3:add/22706-allow-payment-logo?expand=1#diff-71fb9c3c795668ee119208fc1d32456ab71c405962cbc028ae78fe57e5c490c6R118)

**Other Changes**
Due to the pre-commit hook running PHPCS, I was compelled to fix a lot of issues that would have been considered outside the scope of this issue. The majority of these were documenting action/filter hooks.

👉 A new addition that is worth noting is the [nonce check in the Order Data meta box](https://github.com/woocommerce/woocommerce/compare/trunk...johnregan3:add/22706-allow-payment-logo?expand=1#diff-4d6ef31a239517450c07e658510ea400d15787a4d5ace348159994a9e6cd3ed7R606).

### How to test the changes in this Pull Request:

1. In the WC Settings > Payment methods, enter in an html string as the Payment Method title.
Example: `Direct bank transfer <a href="https://wordpress.org" title="WP.org" target="_blank"><img src="https://via.placeholder.com/16/CC0000/" alt="placeholder" title="placeholder" style="border-radius:50%"></a>`
![Screen Shot 2022-06-01 at 11 06 09 AM](https://user-images.githubusercontent.com/2053940/171450839-9f7da3b3-58db-4646-826e-7a462c04cb42.png)

2. Create a New Order, and use the Payment Method set above.
3. After the order status is marked Complete, the image should appear in these places:
  * The Order edit screen, just under the Order title
   
![Screen Shot 2022-06-01 at 11 05 41 AM](https://user-images.githubusercontent.com/2053940/171450068-5976b5b2-62de-48e1-9feb-4bd63ab92a34.png)

  * The Order edit screen, at the order total
  
![Screen Shot 2022-06-01 at 11 05 32 AM](https://user-images.githubusercontent.com/2053940/171450255-fa771f56-4700-4e41-8262-5a89403bf636.png)

  * The User's Checkout page, where they select the Payment Method
  
![Screen Shot 2022-06-01 at 11 07 50 AM](https://user-images.githubusercontent.com/2053940/171450351-ebb38b58-dcca-4109-9a34-10b3e9218295.png)

  * The User's Checkout "Order Received" screen
  
![Screen Shot 2022-06-01 at 11 04 22 AM](https://user-images.githubusercontent.com/2053940/171450413-43585160-77c1-4598-af38-0694d688cfe0.png)

  * The User's My Account > Orders screen
  
![Screen Shot 2022-06-01 at 11 07 20 AM](https://user-images.githubusercontent.com/2053940/171450480-131ba8a7-c4ee-4862-86a7-01ae8c48dc39.png)


### Other information:


### FOR PR REVIEWER ONLY:
